### PR TITLE
pull request

### PR DESCRIPTION
**Pull request by Ajiexwel4**
https://codebreaker2016.herokuapp.com/

### CODE REVIEW

File: lib/modules/marker.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        @game_start = false
+        "Game over! Secret code is #{@secret_code}."
+      else
+        @attempts -= 1
+        '+' * pluses + '-' * (minuses - pluses)
+      end
+    end
+
+    def pluses
+      @player_code.to_s.chars.to_a.map.with_index { |num, index| '+' if num == @secret_code[index] }.compact.size
```
Comment (Alfarlost) [inline lib/modules/marker.rb]: try to stick to max 120 chars per line convention. dat is really hard to read such long lines

File: lib/modules/marker.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    def check_win
+      if win?
+        'Congratulation! You win!'
+      elsif @attempts.zero?
+        @hint = 0
+        @game_start = false
+        "Game over! Secret code is #{@secret_code}."
+      else
+        @attempts -= 1
+        '+' * pluses + '-' * (minuses - pluses)
```
Comment (Alfarlost) [inline lib/modules/marker.rb]: maybe, it is better to incapsulate multiplication logic inside methods pluses and munuses, if you use them only for generating response strings.
e.x. pluses method will return string with right amount of pluseses and the same with minuses.
now they return some random numbers

File: lib/modules/marker.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+module Codebreaker
+  module Marker
+    def check_win
+      if win?
```
Comment (Alfarlost) [inline lib/modules/marker.rb]: to avoid using elseif you can just return congratualtions if win?

File: lib/racker.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      response.set_cookie('guess', param)
+      response.redirect('/')
+    end
+  end
+
+  def save_score_file(name)
+    File.open(path_score_file, 'a') { |file| file.puts "#{name} got #{game.score} score on #{Time.now.asctime}" }
+  end
+
+  def path_score_file
```
Comment (Alfarlost) [inline lib/racker.rb]: there is a small mistake in method naming, should be score_file_path.

**Review by Alfarlost — CHANGES_REQUESTED** [review]
Also some specs, please

File: lib/modules/marker.rb (hunk 3 above)
Comment (Ajiexwel4) [reply lib/modules/marker.rb]: No, I can not. This logic assumed 3 aspects of game: win, game over, marking. It maybe possible if I using nested 'if', but I think that 'elsif' is better.

**Pull request merged**